### ISSUE TITLE
Field size in x/y/z orientations

### ISSUE THREAD

**Issue by semi-h** [xcompact3d/x3d2#24]
We minimize the memory requirement via our memory pool strategy.  We have implemented the allocator module to provide this functionality. Allocator module is responsible from allocating memory on the device or host, and then passing a pointer to a memory region when a method requires a temporary field storage.

Currently, allocator module allocates a 3D array with specified sizes.
`allocate (field(SZ, nx, ny*nz/SZ)`
However, the number of vertices can differ in x, y, and z directions. Therefore, when switching from one orientation to another, or when requesting a pointer from the allocator module, we may need to specify a size so that we have a temporary storage with correct shape. A feature introduced in Fortran 2003 called bounds remapping allows assigning a pointer with a different size or dimension to an existing array and I have tested this feature both with the nvidia and gcc compilers so don't expect any issue here.

To sum up, we need to improve the allocator module a little so that we can specify the shape of the array we need. I think it will be a simple fix to the `get_block` method in the allocator module. This may require us to convert `get_block` from a function to a subroutine. Until this is fixed we can only run simulations where (nx_local == ny_local == nz_local).

I think it also makes sense to allocate a 1D array with right size (nx*ny*nz) when `create_block` is executed because then Fortran compilers should be more comfortable assuming that our pointers point to a continuous piece of memory without any strides.
